server/src/main/java/io/crate/protocols/postgres/types/PGTypes.java

```diff
@@ -72,6 +73,7 @@
         .put(new ArrayType<>(DataTypes.BOOLEAN), PGArray.BOOL_ARRAY)
         .put(new ArrayType<>(DataTypes.TIMESTAMPZ), PGArray.TIMESTAMPZ_ARRAY)
         .put(new ArrayType<>(DataTypes.TIMESTAMP), PGArray.TIMESTAMP_ARRAY)
+        .put(new ArrayType<>(DataTypes.DATE), PGArray.TIMESTAMP_ARRAY)
```

Comment:
Same here. must be a dedicated date array instance.

Reply:
Done. OID is 1182 for pg date array.